### PR TITLE
Add availableFileSystems to save as

### PR DESCRIPTION
**Pull request by alexr00**
Having a Show Local button for Save As is already working, but the suggested filename isn't included in the local path. This change causes the filename to be included as part of the path.

### CODE REVIEW

File: src/vs/workbench/browser/actions/workspaceActions.ts [hunk 1]
```diff
@@ -62,13 +65,20 @@ export class SaveLocalFileAction extends Action {
...
-		@IFileDialogService private readonly dialogService: IFileDialogService
+		@ITextFileService private readonly textFileService: ITextFileService,
+		@IEditorService private readonly editorService: IEditorService
 	) {
 		super(id, label);
 	}
 
 	run(event?: any, data?: ITelemetryData): Promise<any> {
-		return this.dialogService.pickFileToSave({ availableFileSystems: [Schemas.file] });
+		let resource: URI | undefined = toResource(this.editorService.activeEditor);
```
Comment (bpasero) [inline src/vs/workbench/browser/actions/workspaceActions.ts]: Suggest:

```typescript
async run(event?: any, data?: ITelemetryData): Promise<any> {
	let resource: URI | undefined = toResource(this.editorService.activeEditor);
	const options: ISaveOptions = { force: true, availableFileSystems: [Schemas.file] };
	if (resource) {
		return this.textFileService.saveAs(resource, undefined, options);
	}
}
```

File: src/vs/workbench/services/textfile/common/textFileService.ts [hunk 2]
```diff
@@ -648,18 +648,19 @@ export abstract class TextFileService extends Disposable implements ITextFileSer
...
+		return this.fileDialogService.pickFileToSave(this.getSaveDialogOptions(defaultUri, availableFileSystems));
 	}
 
-	private getSaveDialogOptions(defaultUri: URI): ISaveDialogOptions {
+	private getSaveDialogOptions(defaultUri: URI, availableFileSystems?: string[]): ISaveDialogOptions {
 		const options: ISaveDialogOptions = {
 			defaultUri,
-			title: nls.localize('saveAsTitle', "Save As")
+			title: nls.localize('saveAsTitle', "Save As"),
+			availableFileSystems: availableFileSystems,
```
Comment (bpasero) [inline src/vs/workbench/services/textfile/common/textFileService.ts]:
```suggestion
		availableFileSystems,
```

**Review by bpasero — CHANGES_REQUESTED** [review]
@alexr00 some minor code changes suggested.

This action will only ever work with the active file in the editor, is that intended? Can this action be invoked differently, e.g. by right clicking on a file in the explorer?

**Comment by alexr00** [comment]
It is intended that the action behave nearly identically to Save As, with the only difference being that the available file systems are restricted to `Schemas.file`. Since (as far as I can tell) that only applies to the active editor then this new action should also only apply to the active editor.

**Review by bpasero — APPROVED** [review]

**Pull request merged**